netmiko/cisco/cisco_bsp.py

```diff
@@ -0,0 +1,284 @@
+            if re.search(bsp_prompt_pattern, output):
+                log.debug("On BSP Prompt")
+                log.debug("Ctrl + L; press X to enter BMC prompt")
+                self.write_channel(CTRL_L)
+                time.sleep(1 * delay_factor)
+                self.write_channel('x'+self.TELNET_RETURN)
+                time.sleep(1 * delay_factor)
+                output = self.find_prompt()
+
+            if re.search(bmc_prompt_pattern, output):
+                log.debug("On BMC Prompt")
+                return
+
+    def bmc_login(self,prompt_pattern=r'\s*\#',username_pattern='login',pwd_pattern=r'assword',delay_factor=1, max_loops=20):
+        '''
+        Handle BMC login prompt
+
+        :param prompt_pattern: expected BMC prompt
+        :param username_pattern: username prompt pattern
+        :param pwd_pattern: password prompt pattern
+        :param delay_factor: Factor to adjust delays
+        :param max_loops: max number of iterations for attempting to login to BMC prompt before raising exception
+        :return:
+        '''
+
+        bmc_username = "root"
+        bmc_pass = "0penBmc"
+        self.TELNET_RETURN = '\n'
+
+        i = 1
+        while i <= max_loops:
+            i += 1
+            output = self.find_prompt()
+
+            log.debug("Check if we are already on BMC prompt")
+            if re.search(prompt_pattern, output):
+                log.debug("On BMC Prompt")
+                return
+
+            log.debug("Check if BMC Username Prompt detected")
+            if re.search(username_pattern, output):
+                log.debug("BMC Username pattern detected, sending Username={}".format(bmc_username))
+                time.sleep(1)
+                self.write_channel(bmc_username)
+                time.sleep(1 * delay_factor)
+                output = self.find_prompt()
+
+            log.debug("Check if BMC Password Prompt detected")
+            if re.search(pwd_pattern, output):
+                log.debug("BMC Password pattern detected, sending Password={}".format(bmc_pass))
+                self.write_channel(bmc_pass)
+                time.sleep(.5 * delay_factor)
+                output = self.find_prompt()
+
+                if re.search(prompt_pattern, output):
+                    log.debug("On BMC Prompt")
+                    return
+
+                if re.search(pwd_pattern, output):
+                    self.write_channel(bmc_pass)
+                    time.sleep(.5 * delay_factor)
+
+        # Last try to see if we already logged in
+        self.write_channel(self.TELNET_RETURN)
+        time.sleep(.5 * delay_factor)
+        output = self.find_prompt()
+        if re.search(prompt_pattern, output):
+            log.debug("On BMC Prompt")
+            return
+
+        raise NetMikoAuthenticationException("LAST_TRY login failed for BMC Prompt")
+
+    def set_base_prompt(self, pri_prompt_terminator='#',alt_prompt_terminator='$', delay_factor=1):
```

Comment:
i think we need to add a NOT condition for prompt not being left in bmc. safer and easier to debug.

Reply:
and 'bmc' not in prompt ...